### PR TITLE
feat: refactor to use async/await

### PR DESCRIPTION
**Pull request by achingbrain**
Still got a couple of tests failing and some weirdness around the upgrade from `ipfs-api` to `ipfs-http-client`.

~~Depends on ipfs/js-ipfsd-ctl#353~~

### CODE REVIEW

**Comment by achingbrain** [comment]
Refs ipfs/js-ipfs#1670

**Comment by jacobheun** [comment]
I switched CI over to travis (it was still on jenkins) if you want to rebase this.

**Comment by achingbrain** [comment]
Rebased!

**Comment by achingbrain** [comment]
May need ipfs/js-ipfs-http-client/pull/1037

File: package.json [hunk 1]
```diff
@@ -19,17 +19,18 @@
...
-    "ipfsd-ctl": "~0.39.2"
+    "ipfsd-ctl": "~0.44.0"
   },
   "dependencies": {
-    "async": "^2.6.1",
-    "ipfs-api": "^24.0.2",
-    "multiaddr": "^5.0.0",
-    "peer-id": "~0.11.0",
-    "peer-info": "~0.14.1"
+    "ipfs": "^0.36.4",
```
Comment (alanshaw) [inline package.json]: Why does this depend on `ipfs`?

Comment (achingbrain) [reply package.json]: giphy


Oops. Have removed.

**Comment by jacobheun** [comment]
@achingbrain 0.2.3 is out if you want to rebase this bad boy

**Comment by achingbrain** [comment]
Needs ipfs/js-ipfsd-ctl#356

**Comment by achingbrain** [comment]
image

File: src/index.js [hunk 2]
```diff
@@ -60,26 +57,18 @@ class DelegatedContentRouting {
...
-      options = {}
-    } else if (typeof options === 'number') { // This will be deprecated in a next release
-      options = {
-        maxTimeout: options
-      }
-    } else {
-      options = options || {}
-    }
-
+  async * findProviders (key, options = {}) {
```
Comment (jacobheun) [inline src/index.js]: 👍 it will be nice to have an iterable api all ready to go once the dht supports it.

**Comment by jacobheun** [comment]
Released in v0.3.0

File: src/index.js (hunk 2 above)
Comment (achingbrain) [reply src/index.js]: Death to lists.  Iterate all of the things.

Seriously, it's going to alleviate so many memory problems..

**Pull request merged**